src/program.cpp

```diff
@@ -887,6 +887,64 @@ std::string perf_group(instruction_ref ins, bool detailed)
     return result;
 }
 
+void gemm_stats(instruction_ref ins, std::ostream& os)
+{
+    auto op      = ins->get_operator();
+    auto val     = op.to_value();
+    auto op_name = op.name();
+    if(contains(val, "symbol_name"))
+    {
+        op_name = val.at("symbol_name").to<std::string>();
+    }
+
+    if(op_name != "gpu::gemm" and op_name != "gpu::hip_gemm" and op_name != "mlir_dot")
+        return;
+
+    int beta = 0;
+
+    if(contains(val, "beta"))
+    {
+        // from the fuse_ops pass, beta should always equal 0 or 1
+        beta = float_equal(val.at("beta").to<float>(), 1.0) ? 1 : 0;
+    }
+
+    std::vector<instruction_ref> ins_inputs = ins->inputs();
+    auto a_dims                             = ins_inputs.at(0)->get_shape().lens();
+    auto b_dims                             = ins_inputs.at(1)->get_shape().lens();
+    int m                                   = 0;
+    int n                                   = 0;
+    int k                                   = 0;
+    int batch                               = 1;
+    // check for batch dimension
+    if(a_dims.size() == 3)
```

Comment:
There could be multiple batches.

Reply:
right, seems like hip_gemm is restricted to size 2 or 3 but rocblas isn't